pkg/infra/usagestats/service.go

```diff
@@ -70,3 +71,58 @@ func (uss *UsageStatsService) Run(ctx context.Context) error {
 		}
 	}
 }
+
+type memoConcurrentUserStats struct {
+	stats *ConcurrentUsersStats
+
+	memoized time.Time
+}
+
+const concurrentUserStatsCacheLifetime = time.Hour
+
+func (uss *UsageStatsService) GetConcurrentUsersStats(ctx context.Context, mustRefresh bool) (*ConcurrentUsersStats, error) {
+	err := uss.updateConcurrentUsersStatsIfNecessary(ctx, mustRefresh)
+	if err != nil {
+		return nil, err
+	}
+	return uss.concurrentUserStatsCache.stats, nil
+}
```

Comment:
The `mustRefresh` parameter isn't used in practice, I think it ought to be removed, since we shouldn't have inactive code.
```suggestion
func (uss *UsageStatsService) GetConcurrentUsersStats(ctx context.Context) (*ConcurrentUsersStats, error) {
	err := uss.updateConcurrentUsersStatsIfNecessary(ctx)
	if err != nil {
		return nil, err
	}
	return uss.concurrentUserStatsCache.stats, nil
}
```